File: baboon-compiler/src/main/resources/baboon-runtime/java/BaboonExt.java
package baboon.runtime.shared;

public final class BaboonExt {
    private BaboonExt() {}

    public static BaboonDomainVersion domainVersion(BaboonGenerated g) {
        Class<?> klass = g.getClass();
        String identifier = BaboonMetadataAccess.requiredString(klass, "baboonDomainIdentifier");
        String version = BaboonMetadataAccess.requiredString(klass, "baboonDomainVersion");
        return new BaboonDomainVersion(identifier, version);
    }

    public static String baboonUnmodifiedSinceVersion(BaboonGenerated g) {
        return BaboonMetadataAccess.sameInVersions(g.getClass()).get(0);
    }

    public static String unmodifiedSinceVersion(BaboonMeta meta, String typeId) {
        return meta.sameInVersions(typeId).get(0);
    }

}

File: baboon-compiler/src/main/resources/baboon-runtime/cs/BaboonExt.cs
#nullable enable

// ReSharper disable CheckNamespace
// ReSharper disable UnusedMember.Global
// ReSharper disable UnusedType.Global

namespace Baboon.Runtime.Shared
{
    public static class BaboonExt
    {
        public static BaboonDomainVersion DomainVersion(this IBaboonGenerated g)
        {
            return new BaboonDomainVersion(g.BaboonDomainIdentifier(), g.BaboonDomainVersion());
        }

        public static string BaboonUnmodifiedSinceVersion(this IBaboonGenerated g)
        {
            return g.BaboonSameInVersions()[0];
        }

        public static string UnmodifiedSinceVersion(this IBaboonMeta g, string tpe)
        {
            return g.SameInVersions(tpe)[0];
        }
    }
}
